src/Blocks/BlockCrops.h

```diff
@@ -23,11 +23,35 @@ class cBlockCropsHandler :
 	{
 		cFastRandom rand;
 
-		if (a_Meta == 0x7)
+		if (a_Meta == 0x3)
 		{
 			// Is fully grown, drop the entire produce:
 			switch (m_BlockType)
 			{
+				case E_BLOCK_BEETROOTS:
+				{
+					a_Pickups.push_back(cItem(E_ITEM_BEETROOT_SEEDS, static_cast<char>(1 + (rand.NextInt(3) + rand.NextInt(3)) / 2), 0));
+					a_Pickups.push_back(cItem(E_ITEM_BEETROOT, static_cast<char>(1 + (rand.NextInt(3) + rand.NextInt(3)) / 2), 0));  // [1 .. 3] with high preference of 2
+					break;
+				}
+				default:
+				{
+					ASSERT(!"Unhandled block type");
+					break;
+				}
+			}  // switch (m_BlockType)
+		}
+		else if (a_Meta == 0x7)
+		{
+			// Is fully grown, drop the entire produce:
+			switch (m_BlockType)
+			{
+				case E_BLOCK_BEETROOTS:
+				{
+					a_Pickups.push_back(cItem(E_ITEM_BEETROOT_SEEDS, static_cast<char>(1 + (rand.NextInt(3) + rand.NextInt(3)) / 2), 0));
+					a_Pickups.push_back(cItem(E_ITEM_BEETROOT, static_cast<char>(1 + (rand.NextInt(3) + rand.NextInt(3)) / 2), 0));  // [1 .. 3] with high preference of 2
```

Comment:
Sorry @sphinxc0re, it's not bad style, it's needed - we need two (possibly different) random numbers in range 0 .. 2, so we do need two calls to the `NextInt()` function.